pypesto/visualize/optimization_stats.py

```diff
@@ -0,0 +1,353 @@
+    """
+    if properties_to_plot is None:
+        properties_to_plot = ['time', 'n_fval', 'n_grad', 'n_hess', 'n_res',
+                              'n_sres']
+
+    if colors is None:
+        colors = assign_colors_for_list(len(properties_to_plot))
+
+    if legends is None:
+        legends = properties_to_plot
+
+    # TODO: check color and legend shape
+
+    ax = plt.subplots()[1]
+    fig = plt.gcf()
+    fig.set_size_inches(*size)
+
+    for idx, prop_name in enumerate(properties_to_plot):
+        optimization_run_property_per_multistart(
+            results, prop_name, ax, size, start_indices, colors[idx],
+            legends[idx], plot_type)
+
+    ax.set_ylabel("property value")
+    ax.set_title("Optimization properties per optimization run")
+    return ax
+
+
+def optimization_run_properties_per_multistart(
+        results: Union[Result, Sequence[Result]],
+        properties_to_plot: Optional[List[str]] = None,
+        size: Tuple[float, float] = (18.5, 10.5),
+        start_indices: Optional[Union[int, Iterable[int]]] = None,
+        colors: Optional[Union[List[float], List[List[float]]]] = None,
+        legends: Optional[Union[str, List[str]]] = None,
+        plot_type: str = 'line'
+) -> Dict[str, plt.Subplot]:
+    """
+    One plot per optimization property in properties_to_plot.
+
+    Parameters
+    ----------
+    results:
+        Optimization result obtained by 'optimize.py' or list of those
+    properties_to_plot:
+        Optimization run properties that should be plotted
+    size:
+        Figure size (width, height) in inches. Is only applied when no ax
+        object is specified
+    start_indices:
+        List of integers specifying the multistarts to be plotted or
+        int specifying up to which start index should be plotted
+    colors:
+        List of RGBA colors (one color per result in results),
+        or single RGBA color. If not set and one result, clustering is done
+        and colors are assigned automatically
+    legends:
+        Labels for line plots, one label per result object
+    plot_type:
+        Specifies plot type. Possible values: 'line' and 'hist'
+
+    Returns
+    -------
+    ax:
+    The plot axes.
+
+
+    Examples
+    -------
+    optimization_properties_per_multistart(
+        result1,
+        properties_to_plot=['time'],
+        colors=[.5, .9, .9, .3])
+
+    optimization_properties_per_multistart(
+        [result1, result2],
+        properties_to_plot=['time'],
+        colors=[[.5, .9, .9, .3], [.2, .1, .9, .5]])
+
+    optimization_properties_per_multistart(
+        result1,
+        properties_to_plot=['time', 'n_grad'],
+        colors=[.5, .9, .9, .3])
+
+    optimization_properties_per_multistart(
+        [result1, result2], properties_to_plot=['time', 'n_fval'],
+        colors=[[.5, .9, .9, .3], [.2, .1, .9, .5]])
+    """
+
+    if properties_to_plot is None:
+        properties_to_plot = ['time', 'n_fval', 'n_grad', 'n_hess', 'n_res',
```

Comment:
I want to expose it. It can be used if one wants figure per optimization property. Especially if there are multiple results. It will be, e.g. a figure with 'time' stats for all the results, a figure with 'n_fval' stats and so on